Mergin/version_viewer_dialog.py

```diff
@@ -0,0 +1,660 @@
+        self.edit_user_agent.setText(parse_user_agent(self.version_details["user_agent"]))
+        self.edit_user_agent.setToolTip(self.version_details["user_agent"])
+
+        self.model_detail.clear()
+        root_item = QStandardItem(f"Changes in version {self.version_details['name']}")
+        self.model_detail.appendRow(root_item)
+        for category in self.version_details["changes"]:
+            for item in self.version_details["changes"][category]:
+                path = item["path"]
+                item = self._get_icon_item(category, path)
+                if is_versioned_file(path):
+                    if path in self.version_details["changesets"]:
+                        for sub_item in self._versioned_file_summary_items(
+                            self.version_details["changesets"][path]["summary"]
+                        ):
+                            item.appendRow(sub_item)
+                root_item.appendRow(item)
+
+    def _get_icon_item(self, key, text):
+        path = icon_path(self.icons[key])
+        item = QStandardItem(text)
+        item.setIcon(QIcon(path))
+        return item
+
+    def _versioned_file_summary_items(self, summary):
+        items = []
+        for s in summary:
+            table_name_item = self._get_icon_item("table", s["table"])
+            for row in self._table_summary_items(s):
+                table_name_item.appendRow(row)
+            items.append(table_name_item)
+
+        return items
+
+    def _table_summary_items(self, summary):
+        return [QStandardItem("{}: {}".format(k, summary[k])) for k in summary if k != "table"]
+
+    def toggle_project_layers(self, checked):
+        if checked:
+            self.toggle_layers_button.setText("Hide background layers")
+        else:
+            self.toggle_layers_button.setText("Show background layers")
+
+        layers = self.collect_layers(checked)
+        self.update_canvas_layers(layers)
+
+    def update_canvas(self, layers):
+
+        if self.current_diff.isSpatial() == False:
+            self.map_canvas.setEnabled(False)
+            self.save_splitters_state()
+            self.splitter_map_table.setSizes([0, 1])
+        else:
+            self.map_canvas.setEnabled(True)
+            self.set_splitters_state()
+
+        self.update_canvas_layers(layers)
+        self.update_canvas_extend(layers)
+
+    def update_canvas_layers(self, layers):
+        self.map_canvas.setLayers(layers)
+        self.map_canvas.refresh()
+
+    def update_canvas_extend(self, layers):
+        self.map_canvas.setDestinationCrs(QgsProject.instance().crs())
+
+        if layers:
+            extent = layers[0].extent()
+            d = min(extent.width(), extent.height())
+            if d == 0:
+                d = 1
+            extent = extent.buffered(d * 0.07)
+
+            extent = self.map_canvas.mapSettings().layerExtentToOutputExtent(layers[0], extent)
+
+            self.map_canvas.setExtent(extent)
```

Comment:
Not sure what you're trying to do here...
Is it:
```suggestion
            extent = self.map_canvas.mapSettings().layerExtentToOutputExtent(layers[0], extent)
            extent.scale(1.07)
            self.map_canvas.setExtent(extent)
```
?

Reply:
I think the intend is the same is both code, I'm not sure why I some point we defaulted to 1 ?